assets/src/components/map/markers/stopMarker.tsx

```diff
@@ -0,0 +1,305 @@
+import React, { useMemo, useState } from "react"
+
+import { LeafletEventHandlerFnMap, PointTuple } from "leaflet"
+import "leaflet.fullscreen"
+import { MarkerProps } from "react-leaflet"
+
+import { joinClasses } from "../../../helpers/dom"
+import { DirectionId, Stop } from "../../../schedule"
+import { MobileFriendlyTooltip } from "../../mapMarkers"
+import StopCard from "../../stopCard"
+import { ReactMarker } from "../utilities/reactMarker"
+
+/**
+ * Specific variants that the `StopIcon` can render
+ */
+export enum StopType {
+  Small,
+  Medium,
+  Large,
+}
+
+/**
+ * Produces the StopType assigned to the provided map Zoom-Level
+ * @param zoomLevel The Map Zoom Level to use for figuring out the StopType
+ */
+export const stopTypeFromZoomLevel = (zoomLevel: number) => {
+  if (zoomLevel <= 14) {
+    return StopType.Small
+  } else if (15 <= zoomLevel && zoomLevel <= 16) {
+    return StopType.Medium
+  } else if (17 <= zoomLevel) {
+    return StopType.Large
+  } else {
+    throw new Error(
+      "internal error: entered unreachable code: invalid zoom level"
+    )
+  }
```

Comment:
The issue I encountered here was that typescript cannot seem to tell that the entire range is covered by the `if, else if` statements above the `else` so the `else` block should never be reached. So while `17 <= zoomLevel` does cover level's `17` and `18`, I didn't want to decide what a "default" was and what to put in the `else` block because the `else` block should technically never be reached unless there's an improper type passed in or something.